lib/puppet/application/splunk_hec.rb

```diff
@@ -57,9 +57,15 @@ def upload_report(data, _sourcetype)
   end
 
   def main
-    data = JSON.parse(STDIN.read)
+    datainput = STDIN.read
```

Comment:
Is the string read in from STDIN always going to be in the right format? Do we have to wrap this in a `rescue` block to catch a ruby exception on bad input?

Reply:
Latest commit adds a rescue block if we don't get anything on stdin, also checking that we get valid json before beginning to do anything else, so we can inform users of basic error states